### ISSUE TITLE
[FR] Parameter/variable is not used: new pattern

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1277]
Description (screenshot):
--
In the following code, $i could be reported as unused:
```php
function scope(){
    $a = [ 0=> "test", 1=>"test2", 2=>"test3"];
    foreach($a as $i => $value){
        echo "loop";
    }
}
```

Note: even if $value is not used, we can't report it as unused as it is needed in the foreach construct. However, it could be smart to suggest replacing the foreach with a for... when neither the index nor the value are used

Thanks!

**Comment by rentalhost** [comment]
But what is the sense of a looping that not uses `$i` or `$value`?

**Comment by kalessil** [comment]
The native unused inspection has a setting to report unused $i

**Comment by orklah** [comment]
> But what is the sense of a looping that not uses `$i` or `$value`?
To loop a fixed number of times defined by the array's length. But this should be handled by a for() if that's the case



> The native unused inspection has a setting to report unused $i
I missed that one! I stopped reading after the "list()" part and never noticed it after. Thanks!
